File: arm.64/frame/qt/src/find.cpp
#include "type.h"
#include <filesystem>
#include <cstdlib>

namespace fs = std::filesystem;

std::vector<std::string> find(const std::string& target_dir) {
    std::string dir = target_dir;
    if (dir.empty()) {
        const char* env_dir = std::getenv("DIST_DIR");
        if (env_dir && *env_dir) {
            dir = env_dir;
        } else {
            std::vector<std::string> candidates = {
                (fs::current_path() / "dist").string(),
                (fs::current_path() / "../../dist").string(),
                (fs::current_path() / "../../../dist").string(),
                "dist"
            };
            dir = "dist";
            for (const auto& c : candidates) {
                if (fs::exists(c) && fs::is_directory(c)) {
                    dir = fs::canonical(c).string();
                    break;
                }
            }
        }
    }

    std::vector<std::string> files;
    if (!fs::exists(dir) || !fs::is_directory(dir)) {
        return files;
    }

    for (const auto& entry : fs::directory_iterator(dir)) {
        if (entry.is_regular_file() && entry.path().filename() != ".DS_Store") {
            files.push_back(entry.path().string());
        }
    }

    return files;
}

File: arm.64/langs/lua/src/find.lua
local ffi = require("ffi")

local M = {}

function M.find(target_dir)
    local dir = target_dir or os.getenv("DIST_DIR")
    if not dir then
        local candidates = {
            "dist",
            "../../dist",
            "../../../dist"
        }
        for _, c in ipairs(candidates) do
            local f = io.open(c, "r")
            if f then
                f:close()
                dir = c
                break
            end
        end
        if not dir then dir = "dist" end
    end

    local files = {}
    local p = io.popen("ls -1 " .. dir .. " 2>/dev/null")
    if p then
        for file in p:lines() do
            if file ~= ".DS_Store" then
                table.insert(files, dir .. "/" .. file)
            end
        end
        p:close()
    end
    return files
end

return M
